schema/deploy/shipping/views.sql

```diff
@@ -164,4 +164,80 @@ create or replace view shipping.return_results_v1 as
 comment on view shipping.return_results_v1 is
     'View of barcodes and presence/absence results for return of results on website';
 
+
+create or replace view shipping.fhir_encounter_details_v1 as
+
+    with
+        fhir as (
+          select encounter_id,
+                 array_agg(distinct condition."id") as symptoms,
+                 jsonb_object_agg(response."linkId", response.answer) as responses
+            from warehouse.encounter,
+                 jsonb_to_recordset(details -> 'Condition') as condition("id" text),
+                 jsonb_to_recordset(details -> 'QuestionnaireResponse') as q("item" jsonb),
+                 jsonb_to_recordset(q."item") as response("linkId" text, "answer" jsonb)
```

Comment:
Joins to `jsonb_to_recordset` (and other set-returning functions) are implicitly inner lateral joins when used like this. That means if there are no output rows returned by the function (e.g. input is null or empty), then the left hand side input rows will not be included in the final result set. The consequence of this in this view is that a missing piece of FHIR details or questionnaire response will end up removing either responses for related columns or even the entire row for an encounter.

I think this is easiest demonstrated by example:

```sql
with x(a,b) as (
    values
    ('b is null', null::jsonb),
    ('b has an object', '[{"baz": 13}]'),
    ('b has empty object', '[{}]'),
    ('b is empty array', '[]')
)
select * from x, jsonb_to_recordset(b) as b(baz integer);
```
```
         a          │       b       │ baz 
────────────────────┼───────────────┼─────
 b has an object    │ [{"baz": 13}] │  13
 b has empty object │ [{}]          │    
(2 rows)
```

compared to

```sql
select * from x left join jsonb_to_recordset(b) as b(baz integer) on true;
```
```
         a          │       b       │ baz 
────────────────────┼───────────────┼─────
 b is null          │               │    
 b has an object    │ [{"baz": 13}] │  13
 b has empty object │ [{}]          │    
 b is empty array   │ []            │    
(4 rows)
```